src/components/settings/components/Preference.js

```diff
@@ -2,14 +2,13 @@ import React from 'react';
 import { Form,InputNumber,Button,Icon,Modal,Input,Radio,Select,Checkbox,Slider} from 'antd';
 import {languagesArray, timezoneArray} from '../../../common/config/data'
 
-
-
 const Perference = ({
-    form
+    settings, form
   }) => {
-  // const {preference} = settings
+  const {preference} = settings
   function handleChange(type, value) {
-    console.log(type+":"+value);
+    console.log(type+" changed to:"+value);
+    settings.preferenceChange({...preference, [type]: value})
```

Comment:
no need to put other properties in preference

Reply:
fixed